### ISSUE TITLE
[$500] Compose Box - Focus not lost from composer on LHN popup open

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32970]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.11-6
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Open the app
2. Open any report and send any message
3. Focus on composer
4. Click on plus besides the composer to open Actions popup
5. Right on any chat in LHN to open popup on LHN
6. Observe that even though popup is open in LHN, app still focuses on composer

## Expected Result:
App should not focus back on composer whenever any popup is open

## Actual Result:
App focuses back on composer when we open actions popup and open any right click popup in LHN

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/74137c5f-18a4-4e16-8ee5-d8ccfda09552

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0123ed583fdc5774f0</li>
        <li>Upwork Job ID: 1734947870117650432</li>
        <li>Last Price Increase: 2023-12-13</li>
<li>Automatic offers: </li>
<ul>
<li>bernhardoj | Contributor | 28067465</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by mananjadhav** [comment]
I liked @bernhardoj's proposal to fix this with `ComposerWithSuggestions`'s popover context visible state.

🎀 👀 🎀 C+ reviewed

**Comment by mallenexpensify** [comment]
@jasperhuangg when you have a min can you review @bernhardoj 's proposal above?
https://github.com/Expensify/App/issues/32970#issuecomment-1854133465

**Comment by jasperhuangg** [comment]
Sorry for the delay, NewDot deployer chore has taken a lot of time this week. Looks good to me, assigned!

**Comment by bernhardoj** [comment]
@mananjadhav hi, we have a regression [here](https://github.com/Expensify/App/issues/33984). The issue is that the `isPopoverOpen` state is late to update, so the focus always fails. More info is explained on the [proposal](https://github.com/Expensify/App/issues/33984#issuecomment-1877991807) there.

After looking at the linked proposal, I remember we have a [PR](https://github.com/Expensify/App/pull/29199) that will do a big refactor to the composer focus logic and when I look around the PR, I found that they will [handle this issue too](https://github.com/Expensify/App/pull/29199/files#diff-a154cf31ce019f9af207db478cf09e144478a13fa8ee8e67bab5d49fb7e663e9R109-R116).

I don't think there is an easy way to handle this globally, so, sadly I think we should revert our PR and wait for the above PR. (except we want to handle this issue for LHN (report context menu) only, which means my main solution)

Let me know what you think!

**Comment by mallenexpensify** [comment]
@mananjadhav can you review @bernhardoj 's comment above and propose the next best steps? Should I put this on hold pending https://github.com/Expensify/App/pull/29199 ?

**Comment by mananjadhav** [comment]
I can see the PR is already reverted. Yes we should put this on Hold pending #29199

**Comment by mallenexpensify** [comment]
Put on hold pending https://github.com/Expensify/App/pull/29199
Removed 'hold for payment' too

**Comment by anmurali** [comment]
Held, nothing to do here!

**Comment by anmurali** [comment]
https://github.com/Expensify/App/pull/29199 is still being worked on and this issue is on hold.

**Comment by anmurali** [comment]
Still held, started a convo in C+ channel [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1708099567296959)

Also adding a BZ backup for my OOO

**Comment by jasperhuangg** [comment]
Still on HOLD for https://github.com/Expensify/App/pull/29199

**Comment by jasperhuangg** [comment]
Same ^

**Comment by jasperhuangg** [comment]
Same ^^

**Comment by jasperhuangg** [comment]
^^^ Same

**Comment by anmurali** [comment]
Still on hold

**Comment by mallenexpensify** [comment]
Commented on the PR this is held on 
- https://github.com/Expensify/App/pull/29199#issuecomment-2026256884

Also.. are we sure that the behavior isn't intended?  Whenever I click on a chat in LHN, my assumption is that the chat opens and the cursor is in the compose box.  I understand that there was a pop up open previously but we w
ouldn't expect that pop up to stay open if we navigated to another chat then immediately back again, right?  

Maybe there are other examples I'm unaware of where we keep the pop up while taking other actions?

**Comment by mallenexpensify** [comment]
Putting on #vip-vsb for now, since this doesn't have to do with money

**Comment by bernhardoj** [comment]
The issue we have here is that the composer is focused when we open a second popover.

>Click on plus besides the composer to open Actions popup
>Right (click) on any chat in LHN to open popup on LHN
>Observe that even though popup is open in LHN, app still focuses on composer

**Comment by mallenexpensify** [comment]
Thanks @bernhardoj , I def missed that part :ohnothing:
I was able to reproduce
<img width="434" alt="image" src="https://github.com/Expensify/App/assets/22508304/ebf77cda-f1cc-4f83-a4ce-dbe56d93c183">

I feel like it's a pretty rare edge case and it doesn't affect functionality.  BUT... having that second popup from the LHN _hang around_ isn't idea.  Let's wait for this to come off hole then I'll retest then decide what to do with it.

**Comment by jasperhuangg** [comment]
Still on HOLD

**Comment by anmurali** [comment]
Still held

**Comment by anmurali** [comment]
@ntdiary will resume working on https://github.com/Expensify/App/pull/29199
This issue is still blocked on https://github.com/Expensify/App/pull/29199

**Comment by jasperhuangg** [comment]
Still blocked on https://github.com/Expensify/App/pull/29199

**Comment by mallenexpensify** [comment]
- https://github.com/Expensify/App/pull/29199 
Held on above and it looks like it's still a draft

**Comment by jasperhuangg** [comment]
@ntdiary Awesome! Sounds like a good idea to split up the PR. Thanks for the update.

**Comment by mallenexpensify** [comment]
- https://github.com/Expensify/App/pull/42423
Above merged so we're getting close

**Comment by mallenexpensify** [comment]
The PR we were holding on closed without being fixed. I threw a `retest-weekly` on here to see if the bug exists six months later. 
- [Try to refactor the logic regarding refocusing when the modal is closed](https://github.com/Expensify/App/pull/29199#top)

**Comment by jasperhuangg** [comment]
@mallenexpensify Should we keep this as a weekly or demote to monthly if we're gonna wait so far out?

**Comment by jasperhuangg** [comment]
@mallenexpensify @lanitochka17 Are we still retesting this on weekly cadence?

**Comment by mallenexpensify** [comment]
@jasperhuangg retests take a weekish.  @ntdiary and I are both able to reproduce so the bug still exists. I think we need to decide between
1. Close it cuz it's a minor bug that doesn't affect functionality (I might actually prefer to be able to type and have the text fill the compose box vs nothing happening if I type)
2. Review existing proposals to see if we want to accept any
3. Add `Help Wanted` to get more proposals
4. Something. 

I don't think we need/want to increase the price though

**Comment by mallenexpensify** [comment]
Thanks @ntdiary , I'm going to close, comment/reopen if you disagree.
